src/test/util/mod.rs

```diff
@@ -67,43 +68,129 @@ impl std::ops::Deref for TestClient {
     }
 }
 
-impl TestClient {
-    pub(crate) async fn new() -> Self {
-        Self::with_options(None).await
+impl Client {
+    pub(crate) fn test_builder() -> TestClientBuilder {
+        TestClientBuilder {
+            options: None,
+            handler: None,
+            min_heartbeat_freq: None,
+            #[cfg(feature = "csfle")]
+            encrypted: None,
+        }
     }
+}
 
-    pub(crate) async fn with_options(options: impl Into<Option<ClientOptions>>) -> Self {
-        Self::with_handler(None, options).await
+pub(crate) struct TestClientBuilder {
```

Comment:
I realize I'm at risk of the "when the only tool you have is a hammer, every problem looks like a nail" pitfall here, but I think the builder pattern works well for this - it means that the previously-duplicated constructors for accepting permutations of options/additional options/event handler/etc. can be folded into a single implementation.  My motivation for this was being able to construct both `TestClient` and `EventClient` with auto-encryption enabled without having to copy-paste the code across them.

Reply:
Yup, makes sense.  Updated that ticket and left a TODO here.